parastell/nwl_utils.py

```diff
@@ -159,14 +160,27 @@ def flux_coords(vmec, wall_s, coords):
         theta_coords (array of float): poloidal angles of surface crossings
             (rad).
     """
-    phi_coords = np.arctan2(coords[:,1], coords[:,0])
-    theta_coords = []
-    
-    for pt, phi in zip(coords, phi_coords):
-        theta = find_coords(vmec, wall_s, phi, pt)
-        theta_coords.append(theta)
 
-    return phi_coords, theta_coords
+    phi_coords = np.arctan2(coords[:, 1], coords[:, 0])
```

Comment:
netcdf4 datasets do not play nicely with some built in functions that would otherwise be nice for this. What happens here is that the particles are split up according to the number of threads available, then each individual process will load its own instance of the vmec object, loop through the particles assigned, then the results are combined here.

Reply:
So your saying we can't load the vmec object once and then pass it around?

Reply:
Correct. From what I understand, multiprocessing uses pickle, and netcdf4 doesn't pickle apparently, so it can't be passed around by multiprocess. I tried originally to use `multiprocessing.starmap` (which would have been awesome because nothing else would have needed to change) and ran into this issue. Several different forum posts suggested chunking the for loop and loading the data separately for each chunk so I went with that.